### DISCUSSION TITLE
With IsVisible=false, children still evaluated?

### DISCUSSION THREAD

**Discussion by diabhoil** [AvaloniaUI/Avalonia#10866] [Q&A]
Hey I am new to Avalonia (and xaml) and when I debug my app I get these messages:
![image](https://user-images.githubusercontent.com/42711805/229160170-c7e6e609-9779-48f5-b675-74a097173fae.png)

I do not really get why or if they are bad.

My code looks like this
```
<StackPanel 
	IsVisible="{Binding profile.Entry, Converter={x:Static ObjectConverters.IsNotNull}}"
	Grid.Row="2" Orientation="Horizontal">
	<TextBlock 
		Text="{Binding profile.Entry.value1}"
			/>
	<TextBlock 
		Margin="10,0,0,0" 
		Text="{Binding profile.Entry.value2}" />
	<TextBlock
		Text="Test if rendered" />
</StackPanel>
<StackPanel
	IsVisible="{Binding profile.Entry, Converter={x:Static ObjectConverters.IsNull}}"
	Grid.Row="2" Orientation="Horizontal">
	<TextBlock
		Text="No Value"/>
</StackPanel>
```

So I check if profile.Entry is null or not and only render the first StackPanel in case where it is not empty. Rendering is working fine and the first StackPanel is not rendered when the Entry is not null. But why I am still getting these errors when debugging? The TextBlocks in the first StackPanel are still evaluated, even when profileEntry is null and IsVisible should be false? (This component is rendered twice, thats why in the image there are 4 "errors" showing up).

Can you please help my what I am doing wrong here or if there is something wrong at all?

**Comment by timunie** [comment]
I think this issue is related to: https://github.com/AvaloniaUI/Avalonia/issues/10846

IMO, if the App runs as expected, you can ignore the messages for the moment. 

One more hint: Enable compiled bindings if you have not yet. These help in debugging binding related issues a lot. See: https://docs.avaloniaui.net/docs/data-binding/compiledbindings
